scaleway/helpers_baremetal.go

```diff
@@ -171,3 +218,43 @@ func baremetalFindOfferByID(ctx context.Context, baremetalAPI *baremetal.API, zo
 
 	return nil, fmt.Errorf("offer %s not found in zone %s", offerID, zone)
 }
+
+func baremetalCompareOptionIDsToAdd(modifiedOptions, currentOptions []*baremetal.ServerOption, zone scw.Zone) []*baremetal.ServerOption {
+	var toAdd []*baremetal.ServerOption
+
+	m := make(map[string]struct{}, len(currentOptions))
+	for _, option := range currentOptions {
+		m[option.ID] = struct{}{}
+	}
+	// find the differences
+	for _, option := range modifiedOptions {
+		if _, foundID := m[option.ID]; !foundID {
+			toAdd = append(toAdd, option)
+		} else if foundID {
+			if _, foundExp := m[flattenTime(option.ExpiresAt).(string)]; !foundExp {
+				toAdd = append(toAdd, option)
+			}
+		}
+	}
+	return toAdd
+}
+
+func baremetalCompareOptionIDsToDelete(modifiedOptions, currentOptions []*baremetal.ServerOption, zone scw.Zone) []*baremetal.ServerOption {
+	var toDelete []*baremetal.ServerOption
+
+	m := make(map[string]struct{}, len(modifiedOptions))
+	for _, option := range modifiedOptions {
+		m[option.ID] = struct{}{}
+	}
+	// find the differences
+	for _, option := range currentOptions {
```

Comment:
Those two functions look a lot alike. Is there a way to refactor a bit this code to extract common functionality?